contracts/base/strategies/tetu/TetuSwapStrategyBase.sol

```diff
@@ -99,4 +102,136 @@ abstract contract TetuSwapStrategyBase is StrategyBase {
   function emergencyWithdrawFromPool() internal override {
     // noop
   }
-}
+
+  /// @dev Do something useful with farmed rewards
+  function liquidateReward() internal override {
+
+    // assume only xTetu rewards exist
+    address rt = IController(controller()).psVault();
+
+    // it is redirected rewards - PS already had their part of income
+    // in case of pair with xTETU-XXX we not able to separate it
+    uint256 amount = IERC20(rt).balanceOf(address(this));
+    if (amount > 0) {
+      IERC20(rt).safeApprove(_smartVault, 0);
+      IERC20(rt).safeApprove(_smartVault, amount);
+      ISmartVault(_smartVault).notifyTargetRewardAmount(rt, amount);
+    }
+    // first liquidate rewards if bb ratio != 0 and we do not fail previous hardwork after liquidation
+    if (!liquidated && _buyBackRatio != 0) {
+      liquidateRewardTetuSwap();
+    }
+    liquidated = true;
+    // then autocompound the whole remainder
+    autocompoundTetuSwapLP();
+    // set liquidated to false because hardwork is succeeded and next time need to liquidate reward again
+    liquidated = false;
+  }
+
+  function autocompoundTetuSwapLP() internal {
+    address forwarder = IController(controller()).feeRewardForwarder();
+
+    address token0;
+    address token1;
+
+    // define token0 and token1
+    for (uint256 i = 0; i < _rewardTokens.length; i++) {
+      if (_rewardTokens[i] == ITetuSwapPair(pair).token0()) {
+        token0 = _rewardTokens[i];
+      }
+      if (_rewardTokens[i] == ITetuSwapPair(pair).token1()) {
+        token1 = _rewardTokens[i];
+      }
+    }
+
+    uint amountToAddLiquidityToken0 = IERC20(token0).balanceOf(address(this));
+    uint amountToAddLiquidityToken1 = IERC20(token1).balanceOf(address(this));
+
+    // autocompound all that we can liquidate at start
+    if (amountToAddLiquidityToken0 >= 1 && amountToAddLiquidityToken1 >= 1) {
+      IERC20(token0).safeApprove(router, 0);
+      IERC20(token0).safeApprove(router, amountToAddLiquidityToken0);
+      IERC20(token1).safeApprove(router, 0);
+      IERC20(token1).safeApprove(router, amountToAddLiquidityToken1);
+      IUniswapV2Router02(router).addLiquidity(
```

Comment:
no, the main goal of new version add liquidity only once
it is expensive operation and can fail with low amount